modules/developer_manual/pages/testing/unit-testing.adoc

```diff
@@ -153,20 +156,8 @@ OK (251 tests, 724 assertions)
 
 [NOTE] 
 ====
-Apps that are part of core *do not* have their own Makefile; these apps are: 
-
-* comments
-* dav
-* federatedfilesharing
-* federation
-* files
-* files_external
-* files_sharing
-* files_trashbin
-* files_versions
-* provisioning_api
-* systemtags
-* updatenotification.
+Apps that are part of core *do not* have their own Makefile. 
```

Comment:
The "problem" here is that:
1) There are apps whose code is actually in the core repo. Those are the ones listed in the deleted lines. Those apps do not have their own `Makefile`.
2) Then there are apps that are bundled into the core community tarball. 
3) Then there are other supported apps that are not in the default core community tarball.
4) Then there are 3rd-party... miscellaneous apps

(2) and (3) have their own `Makefile` and are the ones that this documentation is about.

(4) might work the same way - it depends what the 3rd-party app developer has done for the tests

Reply:
I'll do my best to summarise this.

Reply:
How about the following?

```asciidoc
Please be aware of the following:

* Apps that are a part of the core repository *do not* have their own Makefile; these are:
** comments
** dav
** federated_file_sharing
** federation
** files
** files_external
** files_sharing
** files_trashbin
** files_versions
** provisioning_api
** system_tags
** update_notification
* Apps that are bundled into the core community tarball *do* have their own Makefile.
* Some apps that are _not_ in the core community tarball *do* have their own Makefile; _and_
* There are 3rd-party, "_miscellaneous_", apps which *may* have their own Makefile.
  These apps are not distributed by https://marketplace.owncloud.com/publishers/owncloud[ownCloud] and are not in xref:installation/apps_supported.adoc[the supported apps list].
  Whether they have one or not depends on the 3rd-party app.
```

Reply:
Reads OK. It is a lot of information when, for running unit tests, the point is just that "maintained by ownCloud" apps have a standardized `Makefile` and so you can follow the instructions here. Other apps may or may not have standardized on the particular `make` command.

The same thing applies to all app setup, JavaScript testing, PHP unit testing, acceptance testing... - if it is a "maintained by ownCloud" app then the documentation should work. Otherwise it may or may not work.

So actually we probably should just have a note on each developer page with a link to the common information about what apps the docs apply to.

Reply:
So, to clarify, this information could be added in a separate page, and then linked from the respective testing pages?

Reply:
Any further feedback, @phil-davis?

Reply:
@phil-davis may I ping...